library/std/src/backtrace.rs

```diff
@@ -443,3 +478,33 @@ impl RawFrame {
         }
     }
 }
+
+#[unstable(feature = "backtrace_frames", issue = "79676")]
+impl<'a> AsRef<[BacktraceFrame]> for Frames<'a> {
+    fn as_ref(&self) -> &[BacktraceFrame] {
+        &self.inner
+    }
+}
+
+
+#[unstable(feature = "backtrace_frames", issue = "79676")]
+impl BacktraceFrame {
+    // Private clone method so that we don't expose a
+    // public BacktraceFrame.clone() by deriving Clone
+    fn clone(&self) -> Self {
+        BacktraceFrame {
+            frame: self.frame.clone(),
+            symbols: self.symbols.clone(),
+        }
+    }
+}
+
+#[unstable(feature = "backtrace_frames", issue = "79676")]
+impl<'a> Iterator for Frames<'a> {
```

Comment:
I think we'll want to avoid adding an `Iterator` impl here just yet, because ideally we'll want to yield `&'a BacktraceFrame`s, but we can't do that until we borrow the frames from the `Backtrace` instead of cloning them.

Reply:
I think in the future it should actually lazily resolve the frames as `.next()` is called to reduce peak memory usage.